client/checkpoint_create.go

```diff
@@ -0,0 +1,13 @@
+package client
+
+import (
+	"github.com/docker/engine-api/types"
+	"golang.org/x/net/context"
+)
+
+// CheckpointCreate creates a checkpoint from the given container with the given name
+func (cli *Client) CheckpointCreate(ctx context.Context, options types.CheckpointCreateOptions) error {
```

Comment:
The signature should be `CheckpointCreate(ctx context.Context, container string, options types.CheckpointCreateOptions)` (and `CheckpointCreateOptions` should not have a `ContainerID` attribute). The container ID or Name is required and thus should not be considered as an option (it make it easier to consume the API too, you know it is required to provide a container Id or Name).